ext/js/pages/settings/languages-controller.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2023-2024  Yomitan Authors
+ * Copyright (C) 2021-2022  Yomichan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import {EventListenerCollection} from '../../core/event-listener-collection.js';
+import {querySelectorNotNull} from '../../dom/query-selector.js';
+
+export class LanguagesController {
+    /**
+     * @param {import('./settings-controller.js').SettingsController} settingsController
+     */
+    constructor(settingsController) {
+        /** @type {import('./settings-controller.js').SettingsController} */
+        this._settingsController = settingsController;
+        /** @type {import('language').Language[]} */
+        this._languages = [];
+        /** @type {EventListenerCollection} */
+        this._eventListeners = new EventListenerCollection();
+    }
+
+    /** */
+    async prepare() {
+        await this._updateOptions();
+        this._languages = await this._settingsController.application.api.getLanguages();
+        this._languages.sort((a, b) => a.iso.localeCompare(b.iso));
+        this._fillSelect();
+    }
+
+    /** */
+    _fillSelect() {
+        const selectElement = querySelectorNotNull(document, '#language-select');
+        for (const {iso, name, flag} of this._languages) {
+            const option = document.createElement('option');
+            option.value = iso;
+            option.text = `(${iso}) ${name} ${flag}`;
```

Comment:
IMO the flags are kinda redundant and they also don't show anything meaningful, and also just show up as text in some cases. I think the language tag is also not helpful since it's just redundant information that is more programmatic than user facing.

![image](https://github.com/themoeway/yomitan/assets/11037431/921d4bbe-ea42-4165-a773-6f8a1d095b65)

As a comparison, here's Firefox's language select:
![image](https://github.com/themoeway/yomitan/assets/11037431/0c623784-bdda-461c-b4fc-d2f613674344)
Where the US is just a language subcategory and not the tag itself.

Reply:
The flags are just eye candy, I find them nice to look at (when they are shown properly).
The language code is displayed because audio for other languages has been handled by [yomichan-forvo-server,](https://ankiweb.net/shared/info/580654285) requiring a custom audio source with the iso code, e.g. `http://localhost:8770?term={term}&language=en`. On yezichak, to make this easier, I also added a {language} token to the custom URLs.

Reply:
What OS/browser are you using that they work in?

Reply:
Chrome on Ubuntu for me. Googling, flags should also work in Firefox on Windows, but not on Windows in general. I'm guessing there's a lot of users on Windows/Chromium where the letters shown may cause confusion with the ISO code in parentheses, so I've removed them (hopefully one day windows also supports them by default).